translator/util/ecsutil/ecsutil.go

```diff
@@ -65,26 +68,29 @@ func (e *ecsUtil) IsECS() bool {
 }
 
 func (e *ecsUtil) getECSMetadata() (em *ecsMetadataResponse, err error) {
-	// choose available endpoint
-	if v3MetadataEndpoint, ok := os.LookupEnv(v3MetadataEndpointEnv); !ok {
-		em, err = e.getMetadataResponse(v2MetadataEndpoint)
-	} else {
+	// Based on endpoint to get ECS metadata, for more information on the respond, https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint.html
```

Comment:
Do we still need to support v2 and v3? If endpoint is there but get metadata fails should we fall back?

Reply:
IDC either way.

Reply:
I think we still need to support v2 and v3 so long as people can create ECS clusters that still rely on them. I think we'd have to investigate further where the drop off points are for v2 and v3 before we could take out support for them.